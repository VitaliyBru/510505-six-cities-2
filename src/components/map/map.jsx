import React, {PureComponent} from "react";
import leaflet from "leaflet";
import PropTypes from "prop-types";

export class Map extends PureComponent {
  constructor(props) {
    super(props);
    this.ICON_URL = `img/pin.svg`;
    this.ICON_SIZE = [30, 30];
  }

  render() {
    return <section className="cities__map map" id="map"/>;
  }

  componentDidMount() {
    const {offersList, offersList: [{city}]} = this.props;
    const cityLocation = [city.location.latitude, city.location.longitude];
    const icon = leaflet.icon({
      iconUrl: this.ICON_URL,
      iconSize: this.ICON_SIZE
    });
    const map = leaflet.map(`map`, {
      center: cityLocation,
      zoom: city.location.zoom,
      zoomControl: false,
      marker: true
    });
    map.setView(cityLocation, city.location.zoom);
    leaflet
      .tileLayer(`https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`, {
        attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
      })
      .addTo(map);

    offersList.forEach((it) => {
      leaflet
        .marker([it.location.latitude, it.location.longitude], {icon})
        .addTo(map);
    });
  }

}

Map.propTypes = {
  "offersList": PropTypes.arrayOf(PropTypes.shape({
    "city": PropTypes.shape({
      "location": PropTypes.shape({
        "latitude": PropTypes.number.isRequired,
        "longitude": PropTypes.number.isRequired,
        "zoom": PropTypes.number.isRequired
      }).isRequired
    }).isRequired,
    "location": PropTypes.shape({
      "latitude": PropTypes.number.isRequired,
      "longitude": PropTypes.number.isRequired
    }).isRequired
  })).isRequired
};
