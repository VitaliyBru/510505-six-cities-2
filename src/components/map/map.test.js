import React from "react";
import renderer from "react-test-renderer";
import {Map} from "./map.jsx";

it(`PlaceList correctly renders after relaunch`, () => {
  const div = global.document.createElement(`div`);
  div.setAttribute(`id`, `map`);
  global.document.body.appendChild(div);

  const tree = renderer
    .create(
        <Map
          cityOffers={[
            {
              "city": {
                "location": {
                  "latitude": 0,
                  "longitude": 0,
                  "zoom": 0
                },
              },
              "location": {
                "latitude": 0,
                "longitude": 0
              },
            }
          ]}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
