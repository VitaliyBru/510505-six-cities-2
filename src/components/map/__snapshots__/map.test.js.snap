// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceList correctly renders after relaunch 1`] = `
<section
  className="cities__map map"
  id="map"
/>
`;
